Classes/Service/Tika/ServerService.php

```diff
@@ -255,15 +289,18 @@ protected function queryTika($endpoint, $context = null)
      * @return string
      * @throws Exception
      */
-    public function extractText(FileInterface $file)
+    public function extractText(FileInterface $file): string
     {
-        $headers = [$this->getUserAgent(), 'Accept: text/plain', 'Content-Type: application/octet-stream', 'Connection: close'];
-
-        $context = stream_context_create(['http' => ['protocol_version' => 1.1, 'method' => 'PUT', 'header' => implode(CRLF, $headers), 'content' => $file->getContents()]]);
+        $request = $this->createRequestForEndpoint('/tika', 'PUT')
+            ->withAddedHeader('Content-Type', 'application/octet-stream')
+            ->withAddedHeader('Accept', 'text/plain')
+            ->withAddedHeader('Connection', 'close')
+            ->withProtocolVersion(1.1)
+            ->withBody($this->convertFileIntoStream($file));
 
-        $response = $this->queryTika('/tika', $context);
+        $response = $this->queryTika($request);
 
-        if ($response === FALSE) {
+        if ($response === false) {
             $this->log('Text Extraction using Tika Server failed', $this->getLogData($file, $response), 2);
```

Comment:
I just recognized that our log level usage is wrong, level `0` and `2`used here are e.g.  `EMERGENCY` and `CRITICAL`. We should check/adapt this afterwards, I will add an issue for that.

Reply:
Instead of setup within the abstract class we should switch to the LoggerAwareInterface and Trait.

Reply:
As a preparation the severity change from integer to the constants of LogLevel (psr)

Reply:
ok, thx

I added an issue for the logging adjustments: #160. As this is not directly related with the PSR-18 client we should plan further adjustments there.